src/npctrade.cpp

```diff
@@ -551,11 +601,17 @@ bool npc_trading::trade( npc &np, int cost, const std::string &deal )
         }
 
         // NPCs will remember debts, to the limit that they'll extend credit or previous debts
-        if( !trade_win.exchange ) {
+        if( ! np.will_exchange_items_freely() ) {
```

Comment:
Btw. the space after the `!` is very unusual. It's not used in the existing code base.

Reply:
I agree that no-space is more common in the CDDA codebase, but a quick grep shows a number of files which have the space-included version, and I think the difference in readability is slight.